Classes/Player/PlayerController.swift

```diff
@@ -146,15 +154,15 @@ class PlayerController: NSObject, Player, PlayerSettings {
 extension PlayerController {
     
     /// Updates the request adapter if it is kaltura type
-    fileprivate func updateRequestAdapterIfExists(forMediaConfig config: MediaConfig) {
+    fileprivate func updateRequestAdapterIfExists(inMediaEntry mediaEntry: MediaEntry) {
         // configure media sources content request adapter if kaltura request adapter exists
         if let _ = self.contentRequestAdapter as? KalturaPlaybackRequestAdapter {
             // create new media session uuid
             self.mediaSessionUUID = UUID()
             // update the request adapter with the updated session id
             self.contentRequestAdapter!.updateRequestAdapter(withPlayer: self)
             // configure media source with the adapter
-            config.mediaEntry.configureMediaSource(withContentRequestAdapter: self.contentRequestAdapter!)
+            mediaEntry.configureMediaSource(withContentRequestAdapter: self.contentRequestAdapter!)
```

Comment:
Not sure what's going on here, because it's not new -- do you change the MediaEntry that was passed by the user? This should not be done.

The request adapter should be saved by AssetBuilder.